index.js

```diff
@@ -3,11 +3,28 @@
 const ndjson = require('ndjson').parse
 const {PassThrough} = require('stream')
 
-const request = require('./lib/request')
+let request;
 
 const isProd = process.env.NODE_ENV === 'production'
 const isObj = o => 'object' === typeof o && !Array.isArray(o)
 
+const configure = (config) => {
+	request = require('./lib/request')(config);
```

Comment:
Don't use global state. Keep `request` defined inside the closure.

Reply:
@derhuerst The problem here is, that I would need to rewrite all the functions, because they rely on a global `request`

Reply:
I'd still do it. Having global state, especially if the API doesn't communicate it, is too dangerous.

You can move all the functions (`departures`, etc) into a `createClient` closure that contains `request`.

Reply:
@derhuerst I didn't like the closure idea, not sure I like my idea any better though :D